### DISCUSSION TITLE
Taskbar+Start menu not working after installing Explorer Patcher

### DISCUSSION THREAD

**Discussion by ghost** [valinet/ExplorerPatcher#3259] [General]
Sorry if this is the wrong place or the wrong way. I'm new here.
Long story short: I had Explorer Patcher working fine on my PC, then I broke my Windows 11 and my system administrator did a fresh install (at least he said he did). Now I'm trying to install Explorer Patcher again. I tried multiple versions all with the same result: some functionalities get applied and are working. However, the windows key on my keyboard doesn't open the start menu anymore, and the buttons on my taskbar don't work. Clicking the windows flag doesn't open Start, clicking the Date+Time does nothing and I don't get the notifications to open.

Anyone who can help me?

![1](https://github.com/valinet/ExplorerPatcher/assets/169458124/275aeb57-868e-4535-b3b7-f0f5b11f8f9c)

**Comment by pyrates999** [comment]
I would not be installing EP on a work computer since it requires admin privileges to install and update.  But if you want to.

If you uninstall EP, does all that work again?

**Reply by ghost** [reply]
Yes, I can correctly uninstall EP and I then get windows 11 taskbar back with functioning start menu, notifications etc.
I suspect it has something to do with something in the registry but that's only my feeling. No hard evidence for that.

**Comment by pyrates999** [comment]
You have a preview release installed: https://support.microsoft.com/en-us/topic/april-23-2024-kb5036980-os-builds-22621-3527-and-22631-3527-preview-5a0d6c49-e42e-4eb4-8541-33a7139281ed

Before installing EP:

Set windows defender to exclude the following:
1. C:\Program Files\ExplorerPatcher
2. %APPDATA%\ExplorerPatcher

Future updates to EP won't be flagged then.

Also set windows defender to exclude the directory that you manually download EP to so you can install it without windows defender blocking it.

Install EP again, and in the EP properties application, go to **About** and click on **Restore default settings**.  Then reboot.

Make sure you install the pre-release here since you have a preview update installed: https://github.com/valinet/ExplorerPatcher/releases/tag/22621.3527.65.3_d398195

Then open up the properties application for explorer patcher, go to advanced, and click on enable console.  And copy and paste the results here when the bug you found has happened.

**Reply by ghost** [reply]
Oke, several things happened / came to my mind:

I did what you described, but clicking on Start or Notifications etc. has zero effect in the console. Nothing appeared.
Only the first time it showed 1 line with something like [EnableXaml] 47ms.
And actually that already made me remember that in the registry I earlier was 'messing' with the DWORD(32-bit) "EnableXamlStartMenu" in Computer\HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced

If you google on **"EnableXamlStartMenu" registry**, you will find some interesting topics. Also about the key StartMenuInit.

My registry is currently as follow: I don't have the DWORD EnableXamlStartMenu at all (I deleted it earlier I guess) and StartMenuInit has value 'd'. Does this has anything to do with it or do you know what values these should have?

Besides that I got some messages from windows security:
![233232](https://github.com/valinet/ExplorerPatcher/assets/169458124/e614f537-5f4d-47ff-86d4-9f9633ce506e)


Finally, I also clicked 'restart file explorer' in EP while console was open. Here is what got logged:

Running on Windows 11, OS Build 10.0.22631.3527.
[Symbols] Symbols for "explorer" are not available.
Attempting to download symbol data; for now, the program may have limited functionality.
Setup user32 functions done
explorer.exe!CImmersiveColor::GetColor() = B666C
explorer.exe!CImmersiveColor::IsColorSchemeChangeMessage() = 15F244
explorer.exe!CImmersiveColorImpl::GetColorPreferenceImpl() = 43E98
explorer.exe!ImmersiveTray::AttachWindowToTray() = 1E9500
explorer.exe!ImmersiveTray::RaiseWindow() = 1EA6E8
explorer.exe!CTaskBand_CreateInstance() = 20BAE4
explorer.exe!HandleFirstTimeLegacy() = 13ACEC
explorer.exe!SetColorPreferenceForLogonUI() = 2A41B8
CAppResolverCacheBuilder::_AddUserPinnedShortcutToStart() = 49240
ITrayUIHost = 3CE308
explorer.exe!TrayUI_CreateInstance() = 1B728C
Setup explorer functions done
Setup uxtheme functions done
[AC] rcMonitorAssignment = 196FC
[AC] blockBegin = 19711
[AC] blockEnd = 1979D
[AC] Patched!
[CC] rcMonitorAssignment = 3D42E7
[CC] rcWorkAssignment = 3D43B4
[CC] blockBegin = 3D42FF
[CC] blockEnd = 3D43CD
[CC] Patched!
[TC] rcMonitorAssignment = 3EC970
[TC] blockBegin = 3EC989
[TC] blockEnd = 3ECA38
[TC] Patched!
[TV] firstCallPrep = 7FEAA
[TV] firstCallCall = 7FEB4
[TV] Patched!
[HC] match1 = 1CD6D
[HC] match2 = 1CE34
[HC] writeAt = 1CA2E
[HC] cleanup = 1CE45-1CE5A
[HC] Patched!
PenMenuSystemTrayManager::GetDynamicSystemTrayHeightForMonitor() = 44FC90
Setup twinui.pcshell functions done
Setup combase functions done
Setup twinui functions done
Setup stobject functions done
Setup bthprops functions done
Setup pnidui functions done
Setup shell32 functions done
Setup windows.storage functions done
[IME] Context menu patch status: 1
Setup inputswitch functions done
Setup peopleband functions done
Installed hooks.
Open Start on monitor thread
EP Service Window thread
Started "Signal shell ready" thread.
Started "Open Start on current monitor" thread.
Initialized taskbar centering module.
[Positioning] Added settings for monitor 0000000000010001 : 1
Registered Win+A, Win+B, and Win+N
Progman: 591842
Progman hook: 11011855
Setup sndvolsso functions done
Ended "Signal shell ready" thread.
0 0
[Updates] Starting daemon.
[Updates] Configured update policy on this system: "Check for updates but let me choose whether to download and install them".
[Updates] Path to module: C:\WINDOWS\dxgi.dll
[Updates] Checking against hash "22622.3527.65.3.dc7569a43a2d446b"
[Updates] Update URL: https://github.com/valinet/ExplorerPatcher/releases/latest/download/ep_setup.exe
[Updates] Hash of remote file is "22622.3527.65.1.0d9ed00606c3e1cb" (valid).
[Updates] Local version obtained from hash is 22622.3527.65.3.
[Updates] No updates are available.
[Symbols] Started "Download symbols" thread.
[Symbols] Attempting to download symbols for OS version 10.0.22631.3527.
[Symbols] Downloading to "C:\Users\WouterRas\AppData\Roaming\ExplorerPatcher\".
[Symbols] Downloading symbols for "C:\WINDOWS\explorer.exe" ("f8f64d8a4a3bb595d2f8947f7e53336c")...
[Symbols] Reading symbols...
[Symbols] Failure in reading symbols for "C:\WINDOWS\explorer.exe".
[Symbols] Finished gathering symbol data.
[Symbols] Finished "Download symbols" thread.


thanks for helping already

**Comment by pyrates999** [comment]
Windows defender flagging it can be ignored.  MS just doesn't like what EP does.

Set windows defender to exclude the following:
1. C:\Program Files\ExplorerPatcher
2. %APPDATA%\ExplorerPatcher

Future updates to EP won't be flagged then.

You can also set windows defender to exclude the directory that you manually download EP to so you can install it without windows defender blocking it.

I also see this:

```
[Symbols] Started "Download symbols" thread.
[Symbols] Attempting to download symbols for OS version 10.0.22631.3527.
[Symbols] Downloading to "C:\Users\WouterRas\AppData\Roaming\ExplorerPatcher".
[Symbols] Downloading symbols for "C:\WINDOWS\explorer.exe" ("f8f64d8a4a3bb595d2f8947f7e53336c")...
[Symbols] Reading symbols...
[Symbols] Failure in reading symbols for "C:\WINDOWS\explorer.exe".
[Symbols] Finished gathering symbol data.
[Symbols] Finished "Download symbols" thread.
```

EP requires the symbols to be downloaded in order for EP to work properly.  Check the EP console output after you've set windows defender to exclude the above directories.  Then restart explorer.exe using the EP properties application with the EP console open to check if the symbols were able to be downloaded.

**Comment by ghost** [comment]
I already let windows defender exclude those locations.
I tried all again but it gave me the same logging.
Don't know why it says Failure in reading symbols for "C:\WINDOWS\explorer.exe".

**Comment by pyrates999** [comment]
Is this a work computer by chance?

**Reply by ghost** [reply]
it's the computer on my office / job yes.

**Reply by pyrates999** [reply]
They might have a firewall at work that is blocking external downloads.  All I can try to suggest is to disable windows defender and disable the windows firewall and see if that fixes it.

**Reply by ghost** [reply]
I can try, but like I said at the beginning: it worked fine at first. So firewall and/or defender weren't a problem the first time.

**Reply by pyrates999** [reply]
MS recently added to defender to detect EP and remove it.  That's why I suggested that.
